### ISSUE TITLE
使用PGnet训练模型的问题，其训练的结果一直下降

### ISSUE THREAD

**Issue by austinleeee** [PaddlePaddle/PaddleOCR#9057]
其训练结果的相应评估指标从下述结果一直下降到0
recall: 0.6122845254606729, precision: 0.5749457157173847, f_score: 0.5930279612223411, seqerr: 1.0, recall_e2e: 0.0, precision_e2e: 0.0, f_score_e2e: 0, fps: 50.335534507290376
![image](https://user-images.githubusercontent.com/116559785/218428370-9cdf408e-1c6f-4474-9f7b-bbab01c3d3e9.png)

- configs：Global:
  use_gpu: True
  epoch_num: 20
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/pgnet_model_seal_16+4pionts/
  save_epoch_step: 10
  eval_batch_step: [ 0, 1000 ]
  cal_metric_during_train: False
  pretrained_model: ./pretrain_models/pgnet_pretrained//best_accuracy
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img:
  infer_visual_type: CN # two mode: EN is for english datasets, CN is for chinese datasets
  valid_set: totaltext # two mode: totaltext valid curved words, partvgg valid non-curved words
  save_res_path: ./output/pgnet_model_seal/predicts_pgnet.txt
  character_dict_path: ./ppocr/utils/ppocr_keys_v1.txt
  character_type: CN
  max_text_length: 50 # the max length in seq
  max_text_nums: 30 # the max seq nums in a pic
  tcl_len: 64

Architecture:
  model_type: e2e
  algorithm: PGNet
  Transform:
  Backbone:
    name: ResNet
    layers: 50
  Neck:
    name: PGFPN
  Head:
    name: PGHead
    character_dict_path: ./ppocr/utils/ppocr_keys_v1.txt # the same as Global:character_dict_path

Loss:
  name: PGLoss
  tcl_bs: 64
  max_text_length: 50 # the same as Global: max_text_length
  max_text_nums: 30 # the same as Global：max_text_nums
  pad_num: 6623# the length of dict for pad

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
    warmup_epoch: 50
  regularizer:
    name: 'L2'
    factor: 0.0001

PostProcess:
  name: PGPostProcess
  score_thresh: 0.5
  mode: fast   # fast or slow two ways
  point_gather_mode: align # same as PGProcessTrain: point_gather_mode

Metric:
  name: E2EMetric
  mode: A   # two ways for eval, A: src from txt,  B: src from gt_mat
  gt_mat_dir:  ./train_data/seal_dataset/  # the dir of gt_mat
  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  main_indicator: f_score_e2e

Train:
  dataset:
    name: PGDataSet
    data_dir: ./train_data/seal_dataset/train
    label_file_list: [./train_data/seal_dataset/train/train_label.txt]
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - E2ELabelEncodeTrain:
      - PGProcessTrain:
          batch_size: 8  # same as loader: batch_size_per_card
          use_resize: True
          use_random_crop: False
          min_crop_size: 24
          min_text_size: 4
          max_text_size: 512
          point_gather_mode: align # two mode: align and none, align mode is better than none mode
      - KeepKeys:
          keep_keys: [ 'images', 'tcl_maps', 'tcl_label_maps', 'border_maps','direction_maps', 'training_masks', 'label_list', 'pos_list', 'pos_mask' ] # dataloader will return list in this order
  loader:
    shuffle: True
    drop_last: True
    batch_size_per_card: 8
    num_workers: 16

Eval:
  dataset:
    name: PGDataSet
    data_dir: ./train_data/seal_dataset/test
    label_file_list: [./train_data/seal_dataset/test/test_label.txt]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - E2ELabelEncodeTest:
      - E2EResizeForTest:
          max_side_len: 768
      - NormalizeImage:
          scale: 1./255.
          mean: [ 0.485, 0.456, 0.406 ]
          std: [ 0.229, 0.224, 0.225 ]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: [ 'image', 'shape', 'polys', 'texts', 'ignore_tags', 'img_id']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 2

- 运行指令/Command Code：
  python tools/train.py -c configs/e2e/e2e_r50_vd_pg.yml
- 训练的是中文印章数据结构：
- seal_dataset/circle_Aug08388.png	[{"points": [[281, 493], [498, 349], [531, 399], [314, 543]], "transcription": "\u68da\u6ede\u72ac", "ignore_tags": [false]}]

**Comment by austinleeee** [comment]
![image](https://user-images.githubusercontent.com/116559785/218432619-a8c6f19b-1787-4ddd-aa5f-9c25b2abc678.png)
![image](https://user-images.githubusercontent.com/116559785/218432744-b2647ddc-471e-461f-98af-8f640bc5d62b.png)
![image](https://user-images.githubusercontent.com/116559785/218432862-cece7b82-0724-48a8-8fb0-b9c1f82e9fff.png)
其loss损失一直在降，评估指标也在下降

**Comment by LDOUBLEV** [comment]
你用的是自己的数据吗，还是公开的数据

**Comment by austinleeee** [comment]
使用的是网上下载的数据，对数据points进行了修改，跟md文档中的印章的数据集对应的
![image](https://user-images.githubusercontent.com/116559785/218664107-d3b4712c-9004-4ac3-9bb3-23999b4381c1.png)
也对上述进行了修改

**Comment by austinleeee** [comment]
> 你用的是自己的数据吗，还是公开的数据

使用的是网上的数据集，对比原md文档修改了points。配置文件如下
![image](https://user-images.githubusercontent.com/116559785/218678094-f7fad67c-c662-4935-ba5e-949a28eec17b.png)
------------------------------
Global:
  use_gpu: True
  ## 最大训练epoch数
  epoch_num: 5
  log_smooth_window: 20
  print_batch_step: 10
  ## 设置模型保存路径
  save_model_dir: ./models/pgnet_model_pre5/
  save_epoch_step: 10
  ## 设置模型评估间隔
  eval_batch_step: [ 0, 1000 ]
  cal_metric_during_train: False
  ## 使用其他的预训练模型(使用印章的)
  pretrained_model: ./pretrain_models/pgnet_r50_vd_seal/best_accuracy
  checkpoints: ./models/pgnet_model_pre5/latest
  save_inference_dir:
  use_visualdl: False
  infer_img: eval_data/test12.png
  infer_visual_type: CN # two mode: EN is for english datasets, CN is for chinese datasets
  valid_set: totaltext # two mode: totaltext valid curved words, partvgg valid non-curved words
  save_res_path: ./output/pgnet_pre5/predicts_pgnet.txt
  character_dict_path: ./ppocr/utils/ppocr_keys_v1.txt
  character_type: CN
  max_text_length: 50 # the max length in seq
  max_text_nums: 30 # the max seq nums in a pic
  tcl_len: 64

Architecture:
  model_type: e2e
  algorithm: PGNet
  Transform:
  Backbone:
    name: ResNet
    layers: 50
  Neck:
    name: PGFPN
  Head:
    name: PGHead
    character_dict_path: ./ppocr/utils/ppocr_keys_v1.txt # the same as Global:character_dict_path

Loss:
  name: PGLoss
  tcl_bs: 64
  max_text_length: 50 # the same as Global: max_text_length
  max_text_nums: 30 # the same as Global：max_text_nums
  pad_num: 6623 # the length of dict for pad

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
    warmup_epoch: 50
  regularizer:
    name: 'L2'
    factor: 0.0001

PostProcess:
  name: PGPostProcess
  score_thresh: 0.5
  mode: fast   # fast or slow two ways
  point_gather_mode: align # same as PGProcessTrain: point_gather_mode

Metric:
  name: E2EMetric
  mode: A   # two ways for eval, A: src from txt,  B: src from gt_mat
  gt_mat_dir:  ./train_data/seal_dataset/  # the dir of gt_mat
  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  main_indicator: f_score_e2e

Train:
  dataset:
    name: PGDataSet
    data_dir: ./train_data/seal_dataset/train
    label_file_list: [./train_data/seal_dataset/train/train_label.txt]
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - E2ELabelEncodeTrain:
      - PGProcessTrain:
          batch_size: 8  # same as loader: batch_size_per_card
          use_resize: True
          use_random_crop: False
          min_crop_size: 24
          min_text_size: 4
          max_text_size: 512
          point_gather_mode: align # two mode: align and none, align mode is better than none mode
      - KeepKeys:
          keep_keys: [ 'images', 'tcl_maps', 'tcl_label_maps', 'border_maps','direction_maps', 'training_masks', 'label_list', 'pos_list', 'pos_mask' ] # dataloader will return list in this order
  loader:
    shuffle: True
    drop_last: True
    batch_size_per_card: 8
    num_workers: 16

Eval:
  dataset:
    name: PGDataSet
    data_dir: ./train_data/seal_dataset/test
    label_file_list: [./train_data/seal_dataset/test/test_label.txt]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - E2ELabelEncodeTest:
      - E2EResizeForTest:
          max_side_len: 768
      - NormalizeImage:
          scale: 1./255.
          mean: [ 0.485, 0.456, 0.406 ]
          std: [ 0.229, 0.224, 0.225 ]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: [ 'image', 'shape', 'polys', 'texts', 'ignore_tags', 'img_id']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 2

**Comment by austinleeee** [comment]
> 其训练结果的相应评估指标从下述结果一直下降到0 recall: 0.6122845254606729, precision: 0.5749457157173847, f_score: 0.5930279612223411, seqerr: 1.0, recall_e2e: 0.0, precision_e2e: 0.0, f_score_e2e: 0, fps: 50.335534507290376 ![image](https://user-images.githubusercontent.com/116559785/218428370-9cdf408e-1c6f-4474-9f7b-bbab01c3d3e9.png)
> 
> * configs：Global:
>   use_gpu: True
>   epoch_num: 20
>   log_smooth_window: 20
>   print_batch_step: 10
>   save_model_dir: ./output/pgnet_model_seal_16+4pionts/
>   save_epoch_step: 10
>   eval_batch_step: [ 0, 1000 ]
>   cal_metric_during_train: False
>   pretrained_model: ./pretrain_models/pgnet_pretrained//best_accuracy
>   checkpoints:
>   save_inference_dir:
>   use_visualdl: False
>   infer_img:
>   infer_visual_type: CN # two mode: EN is for english datasets, CN is for chinese datasets
>   valid_set: totaltext # two mode: totaltext valid curved words, partvgg valid non-curved words
>   save_res_path: ./output/pgnet_model_seal/predicts_pgnet.txt
>   character_dict_path: ./ppocr/utils/ppocr_keys_v1.txt
>   character_type: CN
>   max_text_length: 50 # the max length in seq
>   max_text_nums: 30 # the max seq nums in a pic
>   tcl_len: 64
> 
> Architecture: model_type: e2e algorithm: PGNet Transform: Backbone: name: ResNet layers: 50 Neck: name: PGFPN Head: name: PGHead character_dict_path: ./ppocr/utils/ppocr_keys_v1.txt # the same as Global:character_dict_path
> 
> Loss: name: PGLoss tcl_bs: 64 max_text_length: 50 # the same as Global: max_text_length max_text_nums: 30 # the same as Global：max_text_nums pad_num: 6623# the length of dict for pad
> 
> Optimizer: name: Adam beta1: 0.9 beta2: 0.999 lr: name: Cosine learning_rate: 0.001 warmup_epoch: 50 regularizer: name: 'L2' factor: 0.0001
> 
> PostProcess: name: PGPostProcess score_thresh: 0.5 mode: fast # fast or slow two ways point_gather_mode: align # same as PGProcessTrain: point_gather_mode
> 
> Metric: name: E2EMetric mode: A # two ways for eval, A: src from txt, B: src from gt_mat gt_mat_dir: ./train_data/seal_dataset/ # the dir of gt_mat character_dict_path: ppocr/utils/ppocr_keys_v1.txt main_indicator: f_score_e2e
> 
> Train: dataset: name: PGDataSet data_dir: ./train_data/seal_dataset/train label_file_list: [./train_data/seal_dataset/train/train_label.txt] ratio_list: [1.0] transforms: - DecodeImage: # load image img_mode: BGR channel_first: False - E2ELabelEncodeTrain: - PGProcessTrain: batch_size: 8 # same as loader: batch_size_per_card use_resize: True use_random_crop: False min_crop_size: 24 min_text_size: 4 max_text_size: 512 point_gather_mode: align # two mode: align and none, align mode is better than none mode - KeepKeys: keep_keys: [ 'images', 'tcl_maps', 'tcl_label_maps', 'border_maps','direction_maps', 'training_masks', 'label_list', 'pos_list', 'pos_mask' ] # dataloader will return list in this order loader: shuffle: True drop_last: True batch_size_per_card: 8 num_workers: 16
> 
> Eval: dataset: name: PGDataSet data_dir: ./train_data/seal_dataset/test label_file_list: [./train_data/seal_dataset/test/test_label.txt] transforms: - DecodeImage: # load image img_mode: BGR channel_first: False - E2ELabelEncodeTest: - E2EResizeForTest: max_side_len: 768 - NormalizeImage: scale: 1./255. mean: [ 0.485, 0.456, 0.406 ] std: [ 0.229, 0.224, 0.225 ] order: 'hwc' - ToCHWImage: - KeepKeys: keep_keys: [ 'image', 'shape', 'polys', 'texts', 'ignore_tags', 'img_id'] loader: shuffle: False drop_last: False batch_size_per_card: 1 # must be 1 num_workers: 2
> 
> * 运行指令/Command Code：
>   python tools/train.py -c configs/e2e/e2e_r50_vd_pg.yml
> * 训练的是中文印章数据结构：
> * seal_dataset/circle_Aug08388.png	[{"points": [[281, 493], [498, 349], [531, 399], [314, 543]], "transcription": "\u68da\u6ede\u72ac", "ignore_tags": [false]}]
